sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -388,16 +388,15 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
           partitionColumnNames = partitioningColumns.getOrElse(Nil),
           bucketSpec = getBucketSpec
         )
-        val createCmd = CreateTable(tableDesc, mode, Some(df.logicalPlan))
-        val cmd = if (tableDesc.partitionColumnNames.nonEmpty &&
+        val result = df.sparkSession.sessionState.executePlan(
+          CreateTable(tableDesc, mode, Some(df.logicalPlan)))
```

Comment:
Looks like you want to call `result.toRdd` at here to trigger the execution of a command? Then inside the loop to use toRdd again.